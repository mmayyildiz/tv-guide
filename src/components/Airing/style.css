.tvguide .tvg-grid__program__inner {
    display: inline-flex;
    flex: 0 0 auto;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;
    overflow: hidden;
}

.tvguide .tvg-grid__program__start-time {

    display: flex;
    align-content: center;
    align-items: center;
    margin-top: 10px;
    margin-left: 10px;
    font-weight: 500;
    letter-spacing: .86px;
    opacity: .8;
    font-size: .66rem;
}

.tvguide .tvg-grid__program__title {
    max-height: 40px;
    padding-bottom: 10px;
    padding-left: 10px;
    font-size: .88rem;
}