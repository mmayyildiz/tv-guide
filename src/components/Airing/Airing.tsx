import React, { useState } from 'react';
import Event from '../../models/Event';
import pixelWidth from 'string-pixel-width';
import moment from 'moment';
import AiringDetails from '../AiringDetails/AiringDetails';
import './style.css';

const Airing = (event: Event) => {

    const { st: startTime, d: duration, t: title } = event;

    const [width, setWidth] = useState((duration / 60) * 3);
    const [showPopup, setShowPopup] = useState(false);

    const eventStartTime = moment.utc(startTime);
    const eventEndTime = eventStartTime.add(duration, 'seconds');
    const activeEvent = moment.utc().isBetween(eventStartTime, eventEndTime);
    const eventHour = eventStartTime.format('hh:mm A');


    const onHover = () => {
        const titlePixel = pixelWidth(title, { size: 16, font: 'open sans' });
        if (titlePixel > width) {
            setWidth(titlePixel);
        }
    }

    const onLeave = () => {
        setWidth((duration / 60) * 3);
    }

    return (
        <React.Fragment>
            <div className='tvg-grid__program__inner'
                onMouseEnter={onHover}
                onMouseLeave={onLeave}
                style={{ width: `${width}px`, backgroundColor: activeEvent ? 'grey' : 'white' }}
                onClick={() => setShowPopup(!showPopup)} >
                <span className='tvg-grid__program__start-time'>
                    {eventHour}
                </span>
                <span className='tvg-grid__program__title'>{title}</span>
            </div>

            {showPopup && <AiringDetails event={event} closePopup={() => setShowPopup(!showPopup)} />}
        </React.Fragment>
    );
}

export default Airing;