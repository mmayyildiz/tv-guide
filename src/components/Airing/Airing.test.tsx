import React from 'react';
import { shallow } from 'enzyme';
import Airing from './Airing';

const props = {
    eid: 'foo',
    st: 1,
    d: 1,
    t: 'foo',
    sy: 'foo',
    eg: 'foo',
    s: true,
    ad: true,
    hd: true,
    new: true
};

const mountedComponent = shallow(<Airing {...props} />);

describe('Airing', () => {

    it('should render with props', () => {
        expect(mountedComponent).toMatchSnapshot();

    });

    it('should render the tv guide div ', () => {
        const divs = mountedComponent.find('.tvg-grid__program__inner');
        expect(divs.length).toBe(1);
    });


});
