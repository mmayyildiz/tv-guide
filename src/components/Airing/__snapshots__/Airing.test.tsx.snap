// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airing should render with props 1`] = `ShallowWrapper {}`;
